### PR TITLE
feat: Retry creating wftmpl that references another wftmpl in the same operation

### PR DESCRIPTION
**Pull request by simster7**
Fixes: https://github.com/argoproj/argo-workflows/issues/5173

Looking for feedback for this solution. If feedback is good, then I will move to clean the code, add it to other similar areas (e.g. `ClusterWorkflowTemplates`), and add test.

I initially considered doing topographical sorting on all the workflow templates before attempting to create this. However, that gets a bit messy (have to consider, for example, all steps of all templates of all WorkflowTemplates to look for references). Also, since I expect this use case to come up infrequently, I prefer a "lazy" approach where we only engage this code if we need to.

### CODE REVIEW

File: cmd/argo/commands/template/create.go [hunk 1]
```diff
@@ -69,6 +85,43 @@ func CreateWorkflowTemplates(filePaths []string, cliOpts *cliCreateOpts) {
...
+	// Retry the workflow templates that were not created due to "reference not found" error (see above). This will usually
+	// be empty and not run.
+	for _, wftmpl := range retryWorkflowTemplates {
+		if wftmpl.Namespace == "" {
+			wftmpl.Namespace = client.Namespace()
+		}
+		created, err := serviceClient.CreateWorkflowTemplate(ctx, &workflowtemplatepkg.WorkflowTemplateCreateRequest{
+			Namespace: wftmpl.Namespace,
+			Template:  &wftmpl,
+		})
```
Comment (simster7) [inline cmd/argo/commands/template/create.go:124]: Looking for feedback for this solution. If feedback is good, then I will move to clean the code, add it to other similar areas (e.g. `ClusterWorkflowTemplates`), and add test.

**Comment by alexec** [comment]
I think `withArtifacts` should be a priority over this.

**Comment by simster7** [comment]
> I think withArtifacts should be a priority over this.

It's waiting for review!

**Comment by simster7** [comment]
Closing for now

**Comment by tvalasek** [comment]
hi. this is still an issue in Argo v3.1.12

**Pull request closed without merging**